contracts/contracts/strategies/FraxETHStrategy.sol

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+/**
+ * @title OETH Generalized 4626 Strategy
+ * @notice Investment strategy for vaults supporting ERC4626
+ * @author Origin Protocol Inc
+ */
+import { IERC4626 } from "../../lib/openzeppelin/interfaces/IERC4626.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { IERC20 } from "../utils/InitializableAbstractStrategy.sol";
+import { IWETH9 } from "../interfaces/IWETH9.sol";
+import { IFraxETHMinter } from "../interfaces/IFraxETHMinter.sol";
+import { Generalized4626Strategy } from "./Generalized4626Strategy.sol";
+
+contract FraxETHStrategy is Generalized4626Strategy {
+    using SafeERC20 for IERC20;
+
+    IWETH9 public constant weth =
+        IWETH9(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
+
+    IFraxETHMinter public constant fraxETHMinter =
+        IFraxETHMinter(0xbAFA44EFE7901E04E39Dad13167D089C559c1138);
+
+    function _deposit(address _asset, uint256 _amount) internal override {
+        require(_amount > 0, "Must deposit something");
+        require(
```

Comment:
I think you can add an `else if (_asset == address(assetToken))` below and then an else to revert if not weth or frxETH.